mtp_bank_admin/apps/bank_admin/adi.py

```diff
@@ -0,0 +1,170 @@
+from datetime import datetime
+from collections import defaultdict
+from decimal import Decimal
+
+from django.conf import settings
+from openpyxl import load_workbook, styles
+from openpyxl.writer.excel import save_virtual_workbook
+
+from . import adi_config as config
+from .types import PaymentType, RecordType
+from .exceptions import EmptyFileError
+from .utils import retrieve_all_transactions
+
+
+class AdiJournal(object):
+
+    STYLE_TYPES = {
+        'fill': styles.PatternFill,
+        'border': styles.Border,
+        'font': styles.Font,
+        'alignment': styles.Alignment
+    }
+
+    def __init__(self, payment_type, *args, **kwargs):
+        self.wb = load_workbook(settings.ADI_TEMPLATE_FILEPATH)
+        self.journal_ws = self.wb.get_sheet_by_name(config.ADI_JOURNAL_SHEET)
+
+        self.current_row = config.ADI_JOURNAL_START_ROW
+        self.payment_type = payment_type
+
+    def _next_row(self, increment=1):
+        self.current_row += increment
+
+    def _set_field(self, field, value, style=None, extra_style=None):
+        cell = '%s%s' % (config.ADI_JOURNAL_FIELDS[field]['column'],
+                         self.current_row)
+        self.journal_ws[cell] = value
+
+        if not style:
+            style = defaultdict(dict)
+            style.update(config.ADI_JOURNAL_FIELDS[field]['style'])
+
+        if extra_style:
+            for key in extra_style:
+                style[key].update(extra_style[key])
+
+        for key in style:
+            self.journal_ws[cell].__setattr__(
```

Comment:
wouldn't it be cleaner to use `setattr(obj, name, value)` instead of `__setattr__`?